GitVersionCore/BranchNameComparer.cs

```diff
@@ -0,0 +1,18 @@
+namespace GitVersion
+{
+    using System.Collections.Generic;
+    using LibGit2Sharp;
+
+    class BranchNameComparer : IEqualityComparer<Branch>
+    {
+        public bool Equals(Branch x, Branch y)
+        {
+            return x.Name == y.Name;
```

Comment:
Since we have caseless running this comparison will be case insensitive. Is this correct? If so then the get hashcode should do a lower otherwise the equals does not matc the behaviour a dictionary will exibit